ts/features/payments/bizEventsTransaction/utils/index.ts

```diff
@@ -96,3 +98,36 @@ export const calculateTotalAmount = (
 
   return total.toFixed(2);
 };
+
+export const filterTransactionsByIdAndGetIndex = (
+  transactions: pot.Pot<ReadonlyArray<NoticeListItem>, NetworkError>,
+  transactionId: string
+): {
+  filteredTransactions: Array<NoticeListItem>;
+  removedIndex: number;
+} => {
+  const transactionList = pot.getOrElse(transactions, []);
+  const removedIndex = transactionList.findIndex(
+    transaction => transaction.eventId === transactionId
+  );
+  const filteredTransactions = transactionList.filter(
+    transaction => transaction.eventId !== transactionId
+  );
+  return { filteredTransactions, removedIndex };
+};
+
+export const getTransactionByIndex = (
+  transactions: pot.Pot<ReadonlyArray<NoticeListItem>, NetworkError>,
+  index: number
+): NoticeListItem => pot.getOrElse(transactions, [])[index];
+
+export const restoreTransactionAtIndex = (
```

Comment:
Could it be useful for a jsdoc to explain the purpose of this utility function better?